### DISCUSSION TITLE
Incorrect instruction Ruby

### DISCUSSION THREAD

**Discussion by Pavlusha311245** [nginx/unit#1206] [Support] [answered]
https://unit.nginx.org/howto/samples/#ruby I tried to install the application according to the example, but received an error

Unit config in application section
<img width="713" alt="image" src="https://github.com/nginx/unit/assets/49598333/7bb2794e-7604-42b9-bb8e-b6b0ffd90833">

Ruby file copied from [example](https://unit.nginx.org/howto/samples/#ruby)
<img width="564" alt="image" src="https://github.com/nginx/unit/assets/49598333/af3ba56d-246f-4edf-8d0c-9c616ebddbf2">

Log;:
```text
2024/03/29 08:27:32 [info] 47789#13522582 "ruby" prototype started
2024/03/29 08:27:32 [info] 47790#13522583 "ruby" application started
2024/03/29 08:27:33 [alert] 0#13522583 [unit] Ruby: Failed to require 'rack' package
2024/03/29 08:27:33 [alert] 0#13522583 [unit] Ruby: <internal:/usr/local/Cellar/ruby/3.3.0/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:127:in `require': cannot load such file -- rack (LoadError)
2024/03/29 08:27:33 [alert] 0#13522583 [unit] from <internal:/usr/local/Cellar/ruby/3.3.0/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:127:in `require'
2024/03/29 08:27:33 [notice] 47789#13522582 app process 47790 exited with code 1
2024/03/29 08:27:33 [warn] 47634#13520747 failed to start application "ruby"
2024/03/29 08:27:33 [alert] 47634#13520747 failed to apply new conf
2024/03/29 08:27:33 [notice] 47629#13520721 process 47789 exited with code 0
```

I have 2 guesses. Either the instruction contains an error, or there is an error on the side of the running Unit

Unit version 1.32.1
OS: MacOS 14.3
Modules: Ruby for 1.32.1

**Comment by lcrilly** [comment, marked as answer]
When installing `unit-ruby` for macOS with Homebrew, notice:
```
==> unit-ruby
Make sure rack gem installed:
  /opt/homebrew/opt/ruby/bin/gem install rack
```
So you might need to manually install this dependency. A note in the docs would be useful for macOS.

**Reply by Pavlusha311245** [reply]
Yeah... And you need to restart Unit after that
